src/ipyradiant/visualization/vis_tool.py

```diff
@@ -0,0 +1,114 @@
+from pathlib import Path
+
+import traitlets as T
+
+import ipywidgets as W
+from ipycytoscape import CytoscapeWidget
+from rdflib import Graph, Literal, URIRef
+from rdflib.namespace import RDF
+
+
+class CytoscapeVisualization(W.GridBox):
+    graph = T.Instance(Graph, allow_none=True)
+    cyto_widget = T.Instance(CytoscapeWidget, allow_none=True)
+    nodes = T.List()
+    click_output = T.Instance(W.Output)
+    click_output_box = T.Instance(W.VBox)
+    selected_node = T.Unicode(allow_none=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.edge_color = kwargs.get("edge_color", "pink")
+        self.node_color = kwargs.get("node_color", "grey")
+        self.output_title = W.HTML("<h3>Output from Clicks</h3>")
+        self.click_output_box.children = [self.output_title, self.click_output]
+
+        self.default_edge = {"selector": "edge", "css": {"line-color": self.edge_color}}
+        self.default_node = {
+            "selector": "node",
+            "css": {"background-color": self.node_color},
+        }
+
+        self.children = [self.cyto_widget, self.click_output_box]
+
+    @T.default("click_output")
+    def _make_click_output(self):
+        return W.Output()
+
+    @T.default("nodes")
+    def _make_default_nodes(self):
+        return []
+
+    @T.default("click_output_box")
+    def _make_default_click_output_box(self):
+        return W.VBox()
+
+    @T.default("node_selector_box")
+    def _make_node_selector_box(self):
+        return W.VBox()
+
+    @T.default("cyto_widget")
+    def _make_default_cyto_widget(self):
+        cyto_widget = CytoscapeWidget()
+        cyto_widget.on("node", "click", self.log_node_clicks)
+        cyto_widget.on("edge", "click", self.log_edge_clicks)
+        cyto_widget.set_style([self.default_node, self.default_edge])
+        return cyto_widget
+
+    def log_node_clicks(self, node):
+        with self.click_output:
```

Comment:
i don't see the purpose of this part of the code yet... is this supposed to update `selected_node`(`s`)?

Reply:
this part adds the output info of the node/edge clicked to the output box

Reply:
right, but why is that useful? how will that help be build larger applications that reuse this component? are you expecting people to parse these outputs and do something with it?

Reply:
I don't think it was developed to be included in a final product, but rather as an initial attempt to understand information about nodes and edges. We will need to use the method that Lukas implemented in order to do some processing on selected nodes/edges. 

Once we start to determine what final behavior we want from the first set of initializations I think we can replace it with something more useful, but for now, I think it serves as an example of how we _might_ try to convey information. 

I think it could be a label/text area, but the visualization of the log was just how Lukas did the first implementation.

Reply:
https://github.com/QuantStack/ipycytoscape/blob/master/examples/interaction-example.ipynb

pretty much what Zach said.. im not sure if it will be useful or not, i was just mostly jsut working with the ipycytoscape interaction examples trying to implement something similar (pretty much just copied the code)